common/client-core/gateways-storage/src/lib.rs

```diff
@@ -60,10 +59,10 @@ pub trait GatewaysDetailsStore {
         details: &GatewayRegistration,
     ) -> Result<(), Self::StorageError>;
 
-    async fn upgrade_stored_remote_gateway_key(
+    /// Update the gateway details
+    async fn update_gateway_details(
```

Comment:
this method is quite misleading because you don't overwrite the key in the implementation, meaning if there was a bug somewhere and your new details had a different key, the whole thing would be unusable. imo create a different wrapper for the passed argument, something that doesn't contain the key and **only** has the fields you're actually end up updating.

what you really care about are the listeners, so how about `update_gateway_listeners` or for more futureproofing, `update_published_gateway_data` (or something along those lines)?